logging/test/mock_hub.dart

```diff
@@ -0,0 +1,130 @@
+import 'package:sentry/sentry.dart';
+
+final fakeDsn = 'https://abc@def.ingest.sentry.io/1234567';
+
+class MockHub implements Hub {
```

Comment:
Since we already use `mockito`, would it also be feasible to use it here instead of adding our own mock implementation?

Reply:
I recall us running away from mockito because of a few issues (eg null safety back then), right now we probably have a mix of mockito + fakes, right? ideally, we'd stick to one of them going forward, I'd rather go with the fakes to avoid problems as in the past.

Reply:
Yeah i just ran into issues with `mockito` and and nullability, this can be resolved using (fallback generators)[// https://github.com/dart-lang/mockito/blob/master/NULL_SAFETY_README.md#fallback-generators]. Used this in #643